### DISCUSSION TITLE
Stops added automatically

### DISCUSSION THREAD

**Discussion by Jean-Gabriel1982** [GrandOrgue/GoOdf#87] [Ideas]
Hello Lars, 

when we create our stops in GOodf in the "ranks" section, and we assign them to the windchest corresponding to our keyboards, is it possible, instead of retyping everything, that the names of the stops are automatically created at the same time in the "manuals" or "pedal" section? 

for example in "ranks" I create a Montre 8' which I assign to the Go windchest, automatically this stop will be created in the "manuals">Go>stops" section.

**Comment by larspalo** [comment]
There's no direct odf association between a manual and windchest so that kind of "automatic stop creation" cannot be done from selecting the windchest for a rank (and/or the pipes that actually can be placed on different windchests!).

However, if you don't have a specific need for using "Ranks" you can directly create the "Stop" in the manual instead that has an internal rank. The most important gain from using ranks is re-usability of it among many stops perhaps even on different manuals, if you're not using that feature there's no real gain from using ranks at all.

**Reply by Jean-Gabriel1982** [reply]
thank you for this answer. So if I understand correctly, the Ranks are more useful when you have several samples folders for the same stop, for example for the right side, or the left side of the expressive box , rear or front is that it? 

if we only have one sample (sample folder) per stop, then we should instead use internal rank?

**Reply by larspalo** [reply]
Yes, multi channel versions can be one reason to use ranks instead of the internal rank of a stop, it is not the only possible way to do it but a good example.

There's absolutely nothing wrong with using ranks as such, sometimes it's very useful indeed, but in some other cases it's as un-necessary as "crossing the river for water" and just complicates things. Use what suits your purposes best.
